### ISSUE TITLE
Stop queries after retrieving the required number of results. Add NOCOUNT flag.

### ISSUE THREAD

**Issue by ashtul** [RediSearch/RediSearch#2044]
Currently, if we reach the number of results required by the user, we keep on iterating over the inverted indexes and collect the count of the results. Users may not require this, and we could quit earlier. In some instances, this could save a significant amount of time as we could quit after 10 results instead of after millions of them.
This is true when the user requires no sortby operation.
A `NOCOUNT` flag will quickly exit and reply with the accumulated results to the user.

**Comment by ashtul** [comment]
Related to [this discord question](https://discord.com/channels/697882427875393627/732335216068132995).

**Comment by MeirShpilraien** [comment]
@ashtul Even if the user does not specifically mention sorting we still sort by the score (and we can not change it). I believe that for this feature we will also need some NOSORTING flag, but then it's really the same as FT.AGGREGATE where the user can build his own result processors pipeline so I am not sure why it's needed...

**Comment by ashtul** [comment]
@MeirShpilraien we can remove the sorter if we wish. But for both FT.SEARCH and FT.AGGREGATE we keep on running through the iterators even if we only want 10 results.
This could solve the numeric index issue b/c; if we sort by that field, we can stop after retrieving the required number of results. This is similar to the exercise we have done on the benchmarks.

**Comment by MeirShpilraien** [comment]
This is not true, ft.aggregate stops after reaching the requested number if results (IIRC, and if we do not then its a bug and we should stop, but I believe I am not wrong)

**Comment by MeirShpilraien** [comment]
```
127.0.0.1:6379> FT.CREATE idx SCHEMA t TEXT
OK
127.0.0.1:6379> HSET doc1 t foo
(integer) 1
127.0.0.1:6379> HSET doc2 t foo
(integer) 1
127.0.0.1:6379> FT.SEARCH idx *
1) (integer) 2
2) "doc1"
3) 1) "t"
   2) "foo"
4) "doc2"
5) 1) "t"
   2) "foo"
127.0.0.1:6379> FT.SEARCH idx * limit 0 1
1) (integer) 2
2) "doc1"
3) 1) "t"
   2) "foo"
127.0.0.1:6379> FT.AGGREGATE idx * LOAD 1 t limit 0 1
1) (integer) 1
2) 1) "t"
   2) "foo"
```
You can see that of ft.aggregate you get total number of results 1, because we do not continue when reaching the limit...

**Comment by ashtul** [comment]
@MeirShpilraien I was not convinced so I ran `FT.PROFILE`. It tells the same story you did.
The counter value indeed changes from 2 to 1.
```
127.0.0.1:6379> FT.PROFILE idx search query foo limit 0 1
1) 1) (integer) 2
   2) "doc1"
   3) 1) "t"
      2) "foo"
...
   4) 1) Iterators profile
      2)  1) Type
          2) TEXT
          3) Term
          4) foo
          5) Time
          6) "0.01"
          7) Counter
          8) (integer) 2
          9) Size
         10) (integer) 2
...
127.0.0.1:6379> FT.PROFILE idx aggregate query foo LOAD 1 t limit 0 1
1) 1) (integer) 1
   2) 1) "t"
      2) "foo"
...
   4) 1) Iterators profile
      2)  1) Type
          2) TEXT
          3) Term
          4) foo
          5) Time
          6) "0.0040000000000000001"
          7) Counter
          8) (integer) 1
          9) Size
         10) (integer) 2
```

**Comment by czesio** [comment]
Hi guys. I just want to thanks @ashtul for helping me with this topic.
And I want to say that quite a lot of the time took me to find a solution.  So will be great to put some info clearly to Redis (RediSearch) documentation for user to know that "search" make counting and "agregate" no. And that for big data set searching for phrase that appear in many rows can take significantly longer because of counting.
Thank you.
